src/Platform/Microsoft.Testing.Platform/Hosts/TestHostBuilder.cs

```diff
@@ -243,6 +246,13 @@ public async Task<ITestHost> BuildAsync(
             await testFrameworkCapabilitiesAsyncInitializable.InitializeAsync();
         }
 
+        TestHost.AddDataConsumer(
```

Comment:
Too early here we don't know yet if we're inside the test host or test host controller. 

Add it as last consumer inside the method `BuildTestFrameworkAsync` of this file.